plugins/org.eclipse.fordiac.ide.library/src/org/eclipse/fordiac/ide/library/LibraryManager.java

```diff
@@ -865,36 +872,39 @@ public void resolveDependencies(final IProject project, final TypeLibrary typeLi
 	 * @param linked    set to fill with symbolic names of linked libraries
 	 */
 	private static void findPreferred(final IProject project, final Map<String, Version> preferred,
-			final Set<String> linked) {
-		final IFolder typeLibFolder = project.getFolder(TYPE_LIB_FOLDER_NAME);
-		if (!typeLibFolder.exists()) {
+			final Map<String, IFolder> linked) {
+		final IFolder standardLibFolder = project.getFolder(TypeLibraryTags.STANDARD_LIB_FOLDER_NAME);
```

Comment:
Please make that more generic.
Consider that there might be more virtual folders in the future
Maybe you can make a T[]ypeLibraryTags.getVirtualFolders(...) method

Reply:
Multiple reasons I've not done this:

- 2 folders for linking 2 sources (standard and external libraries)
- making it a list would only add loop overhead
- it's unlikely for any other folders to get added 
- if additional folders were ever to be added, it can be reworked then (going from a simple binary choice to multiple would necessate a redesign anyways)